src/fu-util.c

```diff
@@ -1668,11 +1668,32 @@ fu_util_report_export(FuUtilPrivate *priv, gchar **values, GError **error)
 			return FALSE;
 
 		/* save to local file */
-		filename = g_strdup_printf("%s.json", fwupd_device_get_id(dev));
-		if (!g_file_set_contents(filename, data, -1, error))
+		filename_json = g_strdup_printf("%s.json", fwupd_device_get_id(dev));
+		if (!g_file_set_contents(filename_json, data, -1, error))
 			return FALSE;
 		/* TRANSLATORS: the device ID is printed after this label */
-		g_print("%s %s\n", _("Exported offline report for device:"), filename);
+		g_print("%s %s\n", _("Exported offline report for device:"), filename_json);
+
+		/* self sign data */
+		if (priv->sign) {
+			g_autofree gchar *filename_sig = NULL;
+			g_autofree gchar *sig = NULL;
+
+			sig = fwupd_client_self_sign(priv->client,
+						     data,
+						     FWUPD_SELF_SIGN_FLAG_ADD_TIMESTAMP,
+						     priv->cancellable,
+						     error);
+			if (sig == NULL)
+				return FALSE;
+			filename_sig = g_strdup_printf("%s.p7", fwupd_device_get_id(dev));
```

Comment:
i think `p7b` or `p7c` is the more correct extension -- and it's signing the json, so maybe even `.json.p7c`

Reply:
fixed